tests/test_cli.py

```diff
@@ -225,3 +225,21 @@ def test_run_uncaptured(self, capfd):
             # Check that the expected output is somewhere between the conda logs
             captured = capfd.readouterr()
             assert random_text in captured.out
+
+    def test_run_readonly_env(self):
+        from tests.test_create import make_temp_env
+        from tests.test_create import make_temp_prefix
+        import stat
+
+        prefix = make_temp_prefix(name='test')
+        with make_temp_env(prefix=prefix) as prefix:
+            # Remove write permissions
+            current = stat.S_IMODE(os.lstat(prefix).st_mode)
+            os.chmod(prefix, current & ~stat.S_IWRITE)
+
```

Comment:
Nice, could we add a sanity check here that tries to write a file to the write-protected temp directory?

Reply:
I added a new test here which is failing on windows.  I guess this snippet works on unix but not windows.  Any idea on how to remove the write permissions on windows?

Reply:
Oh my, I now have found this package https://pypi.org/project/oschmod/ which is a wrapper around `win32security` and there's a whole bunch of stuff that has to be done to change file permissions on windows.  So I think I just have to skip this test on windows.

Reply:
Sounds good to me.